src/load/mod.rs

```diff
@@ -20,7 +20,7 @@ pub use stats::Stats;
 pub use status::{RequestStatusProvider, Status, StatusUpdaterThread};
 
 pub fn start_sync(
-    request_generator: impl RequestGenerator + Clone + Send + Sized + 'static,
+    request_generator: impl RequestGenerator + Send + Sized + 'static,
```

Comment:
I think `Sized` is unnecessary here as well, use of the type in the by-value parameter implies `Sized`.

Reply:
Stylistically, this is a better case for old-school generic parameters and a `where` clause.